### PR TITLE
Message "The Cart includes virtual product(s) only, so a shipping add ress is not used."

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (*)
Issue https://github.com/magento/graphql-ce/issues/401

#### Preconditions (*)
1. Registered Customer
2. Virtual Product
3. Developer mode

#### Steps to reproduce (*)
1. Log in as Registered Customer
2. Add Product to Cart
3. Get Customer token
4. Get `ADDRESS_ID` by hovering 'Change Billing Address' or 'Change Shipping Address'
5. Get `CART_ID` by `createEmptyCart` mutation (It returns active `CART_ID`)
6. In GraphQL Client add authorization header and execute

```
mutation {
  setShippingAddressesOnCart(
    input: {
      cart_id: "4P41sMzbNyca3pwMcO4Q5j7m0RNmmvDW"
      shipping_addresses: {
        customer_address_id: 3
        cart_items: {
          cart_item_id: 8
          quantity: 1
        }
      }
    }
  ) {
    cart {
      cart_id
      shipping_addresses {
        address_type
        available_shipping_methods {
          carrier_code
          carrier_title
          method_code
          method_title
          price_excl_tax
          price_incl_tax
        }
        selected_shipping_method {
          amount
          amount
        }
      }
    }
  }
}
```
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php [hunk 1]
```diff
@@ -67,6 +67,11 @@ public function __construct(
      */
     public function execute(ContextInterface $context, CartInterface $cart, array $shippingAddresses): void
     {
+        if ($cart->getIsVirtual()) {
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: We need to move this check on a higher level to `\Magento\QuoteGraphQl\Model\Resolver\SetShippingAddressesOnCart::resolve`
Also. need to cover case with API-functional tests

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by naydav** [comment]
@TomashKhamlai 
Please, verify that we have this case in HipTest
Thanks

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php (hunk 1 above)
Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: Done. add try-catch wrapper and api-functional test

**Comment by naydav** [comment]
Functionality was merged with https://github.com/magento/graphql-ce/pull/330

**Pull request closed without merging**